mimic/rest/auth_api.py

```diff
@@ -44,41 +45,79 @@ def get_token_and_service_catalog(self, request):
 
         tenant_id = (content['auth'].get('tenantName', None) or
                      content['auth'].get('tenantId', None))
+
+        def format_response(callable_returning_session,
+                            nonmatching_tenant_message_generator):
+            try:
+                session = callable_returning_session()
+            except NonMatchingTenantError as e:
+                request.setResponseCode(401)
+                return json.dumps({
+                    "unauthorized": {
+                        "code": 401,
+                        "message": nonmatching_tenant_message_generator(e)
+                    }
+                })
+            else:
+                request.setResponseCode(200)
+                prefix_map = {
+                    # map of entry to URI prefix for that entry
+                }
+
+                def lookup(entry):
+                    return prefix_map[entry]
+                result = get_token(
+                    session.tenant_id,
+                    entry_generator=lambda tenant_id:
+                    list(self.core.entries_for_tenant(
+                         tenant_id, prefix_map, base_uri_from_request(request))),
+                    prefix_for_endpoint=lookup,
+                    response_token=session.token,
+                    response_user_id=session.user_id,
+                    response_user_name=session.username,
+                )
+                result['access']['user']['roles'].append({
```

Comment:
Yep.  I was going to delete it, but figured separate PR.  >:D Unless I can delete it here?